exonum/src/helpers/fabric/details.rs

```diff
@@ -392,6 +406,47 @@ impl GenerateNodeConfig {
 
         (external_address, listen_address)
     }
+
+    /// * `key` - expects EXONUM_SERVICE_PASS or EXONUM_CONSENSUS_PASS.
+    fn get_passphrase(context: &Context, key: &str) -> Vec<u8> {
+        context
+            .get_flag_occurrences(NO_PASSWORD)
+            .map(|_| "".to_string())
+            .or_else(|| env::var(key).ok())
+            .or_else(|| {
+                let stdin = io::stdin();
+                let stdin_locked = stdin.lock();
+                Self::prompt_passphrase(stdin_locked, io::stdout()).ok()
+            }).unwrap()
+            .into_bytes()
+    }
+
+    fn prompt_passphrase<R: BufRead, W: Write>(mut reader: R, mut writer: W) -> io::Result<String> {
+        loop {
+            write!(&mut writer, "Enter passphrase (empty for no passphrase): ")?;
+            let password = rpassword::read_password_with_reader(Some(&mut reader))?;
+            write!(&mut writer, "Enter same passphrase again: ")?;
+            if password == rpassword::read_password_with_reader(Some(&mut reader))? {
+                return Ok(password);
+            }
+            writeln!(&mut writer, "Passphrases do not match. Try again.")?;
+        }
+    }
+
+    fn create_secret_key_file(
+        secret_key_path: impl AsRef<Path>,
+        passphrase: impl AsRef<[u8]>,
+    ) -> PublicKey {
+        let secret_key_path = secret_key_path.as_ref();
+        if secret_key_path.exists() {
+            panic!("{}: File exists", secret_key_path.to_string_lossy(),);
+        } else {
+            if let Some(dir) = secret_key_path.parent() {
+                fs::create_dir_all(dir).unwrap();
+            }
+            generate_keys_file(&secret_key_path, &passphrase).unwrap()
```

Comment:
Slightly off-topic: I'd be wary of using `unwrap()`s: they can leak diagnostic information useful to an adversary.